libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1601,6 +1601,70 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_insert_edge_from_coordinates(int meshKernelId,
+                                                                double firstNodeX,
+                                                                double firstNodeY,
+                                                                double secondNodeX,
+                                                                double secondNodeY,
+                                                                int& firstNodeIndex,
+                                                                int& secondNodeIndex,
+                                                                int& edgeIndex)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+
+            std::unique_ptr<meshkernel::CompoundUndoAction> compoundUndoAction = meshkernel::CompoundUndoAction::Create();
+            meshkernel::Point const firstNodeCoordinates{firstNodeX, firstNodeY};
+            meshkernel::Point const secondNodeCoordinates{secondNodeX, secondNodeY};
+
+            const auto searchRadius = ComputeDistance(firstNodeCoordinates, secondNodeCoordinates, meshKernelState[meshKernelId].m_projection) * 0.1;
+
+            if (searchRadius <= 0.0)
+            {
+                throw meshkernel::MeshKernelError("The nodes are coinciding.");
+            }
+
+            meshKernelState[meshKernelId].m_mesh2d->BuildTree(meshkernel::Location::Nodes);
+            auto firstNodeId = meshKernelState[meshKernelId].m_mesh2d->FindNodeCloseToAPoint(firstNodeCoordinates, searchRadius);
+            if (firstNodeId == meshkernel::constants::missing::uintValue)
+            {
+                auto result = meshKernelState[meshKernelId].m_mesh2d->InsertNode(firstNodeCoordinates);
+                firstNodeId = std::get<0>(result);
```

Comment:
You may get a sonarcloud warning here to use structured binding. 
Could use std::tie